packit/config/package_config.py

```diff
@@ -254,6 +254,151 @@ def __eq__(self, other: object):
         )
```

Comment:
You are adding too much new classes with no docstrings - for me it's easy to get lost there: since the 4 new classes accept PackageConfig in a constructor, I'd personally turn them into a single one. On the other hand, feel free to explain what's your thinking process for the current design and we can discuss.